src/GitHub.App/Controllers/UIController.cs

```diff
@@ -142,6 +136,27 @@ public IObservable<UserControl> SelectFlow(UIControllerFlow choice)
             return transition;
         }
 
+        /// <summary>
+        /// Allows listening to the completion state of the ui flow - whether
+        /// it was completed because it was cancelled or whether it succeeded.
+        /// </summary>
+        /// <returns>true for success, false for cancel</returns>
+        public IObservable<bool> ListenToCompletionState()
+        {
+            if (completion == null)
```

Comment:
As a stylistic thing, I tend to like:

```csharp
completion = completion ?? new Subject<bool>();
```

But it's a personal preference. I'll still merge this.